editor/project_manager/project_dialog.cpp

```diff
@@ -245,6 +245,38 @@ void ProjectDialog::_validate_path() {
 			}
 		}
 	}
+
+	// Check if the target path is a subdirectory of original when duplicating
+	if (mode == MODE_DUPLICATE) {
+		String base_path = original_project_path;
+		String duplicate_target = target_path;
+
+		// Ensure the paths end with a slash
+		if (!base_path.ends_with("/")) {
+			base_path += "/";
+		}
+
+		if (!duplicate_target.ends_with("/")) {
+			duplicate_target += "/";
+		}
+
+		bool is_subdirectory;
+		bool is_equal;
+
+		if (DirAccess::is_case_sensitive()) {
+			bool is_subdirectory = duplicate_target.begins_with(base_path);
+			bool is_equal = base_path == duplicate_target;
```

Comment:
If strings are equal, they also begin with each other, so the second check is redundant.